packages/native-stack/src/types.tsx

```diff
@@ -452,6 +454,43 @@ export type NativeStackNavigationOptions = {
    * @platform ios
    */
   gestureResponseDistance?: ScreenProps['gestureResponseDistance'];
+  /**
+   * Changes the gesture for dismissing the screen to the previous one.
+   *
+   * Supported values:
+   * - "swipeRight": swipe right to dismiss the screen
+   * - "swipeLeft": swipe left to dismiss the screen
+   * - "swipeUp": swipe up to dismiss the screen
+   * - "swipeDown": swipe down to dismiss the screen
+   * - "verticalSwipe": swipe up or down to dismiss the screen
+   * - "horizontalSwipe": swipe left or right to dismiss the screen
+   * - "twoDimensionalSwipe": swipe in any direction to dismiss the screen
+   *
+   * Only supported on iOS and Android.
+   */
+  goBackGesture?: GoBackGesture;
```

Comment:
I'm curious how we should name it then. `gestureToTraverseBack`? `gestureToPreviousPath`? @satya164 @piaskowyk do you have any ideas?

Reply:
@tboba how about `gestureType`? since it specifies the type of gesture to use. or something similar.

I think `TraverseBack`, `PreviousPath` etc. aren't necessary since that's what all of the gesture related options are for, so it'd be redundant.

Reply:
> @tboba What about gestureToGoBack / gestureBack

But all of the gesture related props are for going back as I mentioned in both of my replies. Is there a gesture related prop that's not for going back? If we're not using the "back" terminology anywhere else and it's not confusing there, why would it be confusing here?

> Because you know, gestureType for what

You can ask the same question for all of the other `gestureX` options.

Reply:
Another name could be `gestureDirection` since all of these props seem to be about direction of the gesture (and we already have such a prop on JS stack):

```ts
type GestureDirection = 'left' | 'right' | 'up' | 'down' | 'horizontal' | 'vertical' | 'any';
```

Reply:
Sure, you've convinced me 👍 However, I'm afraid about the inconsistency of passed values for both `gestureDirection` on JS stack and `goBackGesture` (for now) on native-stack, since they both could accept different props, even after making that prop accept strings, as of the idea you've mentioned below.

Reply:
@tboba v7 of stack is not stable yet, so there's still a chance to align it

Tho I'm also curious, what happens to `left` and `right` in RTL mode? are they flipped?

Reply:
@satya164 for now we can't align accepted values to these given from JS stack, since this type is taken from reanimated to avoid strong dependency. This would require us to wait for a pretty long chain from releasing a new version of reanimated (new values would be also a breaking change there) to have that version accepted on Expo Go in new release. 

Also, the gesture on RTL mode is not reverted, but we can fix this out of this PR.